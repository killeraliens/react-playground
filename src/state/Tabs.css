/*.Tabs {
  display: flex;
  flex-direction: row;
}
*/
/*.Tabs button {
  display: inline-block;
}

.Tabs div {
  display: block;
}*/
