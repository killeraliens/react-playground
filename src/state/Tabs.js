import React from 'react';
import './Tabs.css';


class Tabs extends React.Component {
  static defaultProps = {
    tabs: []
  }

  constructor(props) {
    super(props);

    this.state = {
      current: 0
    }
  }

  handleButtonClick(index) {
    this.setState({current: index});
  }

  renderButtons() {
    return this.props.tabs.map((tab, i) => (
      <button key={i} id={i} onClick={() => this.handleButtonClick(i)}>{tab.name}</button>
    ));

  }

  renderContent() {
    const currentTab = this.props.tabs[this.state.current];
    return(
      <div className='tab-content'>
        {currentTab.content}
      </div>
    )
  }

  render() {
    return (
      <div className='Tabs'>
        {this.renderButtons()}
        {this.props.tabs[0] && this.renderContent()}
      </div>
    )
  }
}

export default Tabs;
