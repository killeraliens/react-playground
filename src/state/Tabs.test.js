import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs from './Tabs';


const tabsProp = [
  {name: 'name 1', content: 'content 1 & some content'},
  {name: 'name 2', content: 'content 2 & some content'},
  {name: 'name 3', content: 'content 3 & some content'}
];


describe('Tabs Component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Tabs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a button for each tab', () => {
    const wrapper = shallow(<Tabs tabs={tabsProp} />);
    const buttonQty = wrapper.find('button').length;
    expect(buttonQty).toEqual(tabsProp.length);
  });

  it('button text should match tab item title', ()=> {
    const wrapper = shallow(<Tabs tabs={tabsProp} />);
    const firstButtonText = wrapper.find('button').at(0).innerText;
    expect(firstButtonText).toEqual(tabsProp[0].title);
  });

  it('shows first tab content on default', () => {
    // const div = document.createElement('div');
    // ReactDOM.render(<Tabs tabs={tabsProp} />, div);
    // ReactDOM.unmountComponentAtNode(div);
    const wrapper = shallow(<Tabs tabs={tabsProp} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });


})
