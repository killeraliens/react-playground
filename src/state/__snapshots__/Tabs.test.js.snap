// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders empty with no tab props passed 1`] = `
<div
  className="Tabs"
/>
`;

exports[`Tabs Component shows first tab content on default 1`] = `
<div
  className="Tabs"
>
  <button
    id={0}
    key="0"
    onClick={[Function]}
  >
    name 1
  </button>
  <button
    id={1}
    key="1"
    onClick={[Function]}
  >
    name 2
  </button>
  <button
    id={2}
    key="2"
    onClick={[Function]}
  >
    name 3
  </button>
  <div
    className="tab-content"
  >
    content 1 & some content
  </div>
</div>
`;
